### PR TITLE
HWKAGENT-158 Collection Interval Changes

### PR DESCRIPTION
**Pull request by jshaughn**
- add new subsystem configuration for minimum collection interval
  - default to 30s (changed from hardcode of 1s)
- let collection interval 0 indicate collection is disabled
- consolidate xml and Protocol service on 600s auto-discovery

@jmazzitelli You can take a look as is to see if it looks kosher.  I didn't add any tests yet, I'll look at that on Monday.

### CODE REVIEW

**Comment by jmazzitelli** [comment]
LGTM

**Comment by jshaughn** [comment]
@jmazzitelli Added an itest (and a little itest infra) but it's not working yet.  It's going to need the full agent restart stuff to work, triggered when updating a metric/avail attribute.  So, waiting on that PR and then I'll rebase and retest.

**Comment by jshaughn** [comment]
@jmazzitelli I rebased on master and I've updated AgentInstallerITest to test for a disabled metric (and implicitly a collection interval update restarting the agent).  I think is ready for merge.

File: hawkular-wildfly-agent-itest-parent/hawkular-wildfly-agent-itest-util/src/main/java/org/hawkular/wildfly/agent/itest/util/AbstractITest.java [hunk 1]
```diff
@@ -222,6 +222,31 @@ protected String getNodeAttribute(ModelControllerClient mcc, ModelNode addressAc
...
+    protected void writeNodeAttribute(ModelControllerClient mcc, ModelNode addressActual, String attributeName,
+            Integer value) {
+        OperationBuilder.writeAttribute()
+                .address(addressActual)
+                .attribute(attributeName, value)
+                .execute(mcc)
+                .assertSuccess();
+    }
+
+    protected void restartAgent() {
```
Comment (jmazzitelli) [inline hawkular-wildfly-agent-itest-parent/hawkular-wildfly-agent-itest-util/src/main/java/org/hawkular/wildfly/agent/itest/util/AbstractITest.java]: do you still need this restartAgent() method? Get rid of it if not needed anymore.

Comment (jshaughn) [reply hawkular-wildfly-agent-itest-parent/hawkular-wildfly-agent-itest-util/src/main/java/org/hawkular/wildfly/agent/itest/util/AbstractITest.java]: yeah, not called anymore, I'll remove.

**Comment by jshaughn** [comment]
Hold On, checking one thing...

**Pull request merged**